sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import numpy as np
+
+# class Polygon:
+
+
+def get_centre_of_grasp(finger_tips):
+    """
+    Calculate the centre of the grasp
+    """
+    n = len(finger_tips)
+    sum_x = 0.0
+    sum_y = 0.0
+    sum_z = 0.0
+
+    for keys, values in finger_tips.items():
+        x, y, z = values
+        sum_x += x
+        sum_y += y
+        sum_z += z
+
+    return (sum_x/n), (sum_y/n), (sum_z/n)
+
+
+def find_cosine_angle(a, b, c):
+    """ NOT TESTED """
+    ba = a - b
+    bc = c - b
+
+    cosine_angle = np.dot(ba, bc) / (np.linalg.norm(ba) * np.linalg.norm(bc))
+    angle = np.arccos(cosine_angle)
+
+    return angle  # np.degrees(angle)
+
+
+def main():
+    """ TEST for find_triangle_area """
+    a = np.array([0.0, 0.0, 1.0])
+    b = np.array([2.0, 3.0, 1.0])
+    c = np.array([3.0, 0.0, 1.0])
+    print find_triangle_area(a, b, c)
+
+
+def find_triangle_area(a, b, c):
+    """
+    Calculates the area of a triangle given the points at its corners
+    """
+    ab = a - b  # np.dot(a, b)
+    ac = a - c  # np.dot(a, c)
+
+    cross_product = np.cross(ab, ac)
+
+    triangle_area = np.linalg.norm(cross_product) / 2
+
+    return triangle_area
+
+# ''' ALTERNATIVE: Have not yet got working correctly '''
+# def find_triangle_area(a, b, c):
+#     ab = abs(a - b)  # np.dot(a, b)
+#     ac = abs(a - c)  # np.dot(a, c)
+#     print ab, ac
+#
+#     triangle_area = 0.2 * np.sqrt((ab[1]*ac[2] - ab[2]*ac[1])**2 + (ab[2]*ac[0] - ab[0]*ac[2])**2 +
+#                                   (ab[0]*ac[1] - ab[1]*ac[0])**2)
+#     return triangle_area
+
+
+def poly_3d_area(finger_tips):
+    """
+    Polygon area in 3D for Hand E (Five fingers)
+    """
+    polygon_area = 0.0
+
+    # 'rh_fftip', 'rh_mftip', 'rh_rftip',    'rh_lftip', 'rh_thtip'
+
+    xa, ya, za = finger_tips['rh_fftip']  # 1st finger-tip
```

Comment:
Please see my other comment. Let's not assume fixed names for these links (we want this to be compatible with hand H)